src/composables/coinTokensProps.ts

```diff
@@ -28,6 +40,12 @@ export function useTokenProps() {
     isMultisig: tokenProps.value?.isMultisig ?? false,
   }));
 
+  watch(route, () => {
+    if (route.params.id !== tokenProps.value?.contractId) {
+      tokenProps.value = null;
+    }
+  }, { immediate: true });
```

Comment:
I noticed that we have performance issues with watcher initiated multiple time every time the composable is fired. Please bare in mind that every time `useTokenProps` is fired the watcher is created once again. This is why I suggest to create `let initialized` before the composable and run watchers only once. Here's an example: https://github.com/superhero-com/superhero-wallet/pull/2598/files#diff-1698ea5d81d976b7229ebb41ba6b715373125bdc644776ba7774792ffbefd7e9R256

Reply:
But to be honest I think this composable should not have watcher at all. The component that loads it should take care of resetting the state.